bundles/NewsletterBundle/src/PimcoreNewsletterBundle.php

```diff
@@ -0,0 +1,80 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\NewsletterBundle;
+
+use Pimcore\Bundle\AdminBundle\Support\BundleAdminSupportTrait;
+use Pimcore\Bundle\AdminBundle\Support\PimcoreBundleAdminSupportInterface;
+use Pimcore\Bundle\NewsletterBundle\DependencyInjection\Compiler\CustomReportsPass;
+use Pimcore\Extension\Bundle\AbstractPimcoreBundle;
+use Pimcore\Extension\Bundle\Traits\PackageVersionTrait;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class PimcoreNewsletterBundle extends AbstractPimcoreBundle implements PimcoreBundleAdminSupportInterface
+{
+    use BundleAdminSupportTrait;
+    use PackageVersionTrait;
+
+    public function getJsPaths(): array
+    {
+        return [
+            '/bundles/pimcorenewsletter/js/startup.js',
+            '/bundles/pimcorenewsletter/js/document/newsletter.js',
+            '/bundles/pimcorenewsletter/js/document/newsletters/settings.js',
+            '/bundles/pimcorenewsletter/js/document/newsletters/sendingPanel.js',
+            '/bundles/pimcorenewsletter/js/document/newsletters/plaintextPanel.js',
+            '/bundles/pimcorenewsletter/js/document/newsletters/addressSourceAdapters/default.js',
+            '/bundles/pimcorenewsletter/js/document/newsletters/addressSourceAdapters/csvList.js',
+            '/bundles/pimcorenewsletter/js/document/newsletters/addressSourceAdapters/report.js',
```

Comment:
the custom report support works fine now, but the option still appears even if the bundle is disabled due to this js

Reply:
not sure if is possible to selectively load from `getJsPaths()`

what comes up to my mind is to add a property like
`requiredNamespace: ['pimcore.bundle.customreports']`
under `report.js`

and check if the bundle namespace exists? 
https://github.com/pimcore/pimcore/blob/12d9fcaa61cc9e9e0e4414c12d3faeb7518c72ca/bundles/NewsletterBundle/public/js/document/newsletters/sendingPanel.js#L83-L84 
`pimcore.bundle.newsletter.document.newsletters.addressSourceAdapters[adapterNames].requiredNamespace` or so?

Reply:
Used property here https://github.com/pimcore/pimcore/pull/14848/commits/b38378255b25bf20f17b6871f82bb98e10b5ca86